src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy

```diff
@@ -3112,15 +3125,19 @@ public enum LegendsAwakened implements LogicCardInfo {
             energyCost ()
             attackRequirement {}
             onAttack {
-              damage 0
+              assert my.bench : "Your bench is empty"
+              my.bench.each{
+                heal 30, it
+              }
             }
           }
           move "Supreme Blast", {
-            text "200 damage. in play, this attack does nothing. Discard all Energy attached to Mesprit."
+            text "200 damage. If you don't have Uxie LV.X and Azelf LV.X in play, this attack does nothing. Discard all Energy attached to Mesprit."
             energyCost P, P
-            attackRequirement {}
+            attackRequirement {my.all.findAll{it.name == ("Uxie Lv.X")} && my.all.findAll{it.name == ("Uxie Lv.X")}}
             onAttack {
-              damage 0
+              damage 200
+              discardAllEnergySelf()
```

Comment:
The order of Energy and Self is still incorrect. `discardAllEnergySelf` may make more sense using english rules, but `discardAllSelfEnergy` is actually in the codebase.